packages/core/docs/commercetools/composables/use-forgot-password.md

```diff
@@ -0,0 +1,98 @@
+# `useForgotPassword`
+
+## Features
+
+`useForgotPassword` composable can be used to:
+
+* generate reset password token
+* change user password using token
+
+## API
+
+- `reset` - function used to generate reset password token. When invoked, it requests data from the API and populates `token` property. This method accepts a single params object. The `params` has the following option:
+
+  - `params: IResetPasswordParams`
+
+  - `customQuery?: CustomQuery`
+
+```typescript
+interface IResetPasswordParams {
+    email: string;
+}
+
+type CustomQuery = {
+  customerCreatePasswordResetToken: string
+}
+```
+
+- `change` - function used to set new user password after reset. When invoked, it requests data from the API and populates `result` property. This method accepts a single params object. The `params` has the following option:
+
+  - `params: IChangePasswordParams`
+  
+  - `customQuery?: CustomQuery`
+
+```typescript
+interface IChangePasswordParams {
+    tokenValue: string;
+    newPassword: string;
+}
+
+type CustomQuery = {
+  customerResetPassword: string
+}
+```
+
+- `token: string` - reactive data string containing the created token.
+
+- `loading: boolean` - reactive object containing information about loading state of `change` and `reset` methods.
+
+- `error: UseForgotPasswordErrors` - reactive object containing the error message, if `change` or `reset` failed for any reason.
+
+```ts
+interface UseForgotPasswordErrors {
+  result: Error;
+}
+```
+
+## Getters
+
+We do not provide getters for forgotPassword and its parts.
+
+## Example
+
+Generating reset password token and changing user password.
+
+```vue
+<template>
+  <!-- Generate reset password token -->
+  <form @submit.prevent="reset({ email: form.value.email })">
+    <!-- form fields -->
+    <button type="submit" :disabled="loading">Reset Password</button>
+  </form>
+
+  <!-- Change user password -->
+  <form @submit.prevent="change({ tokenValue: <TOKEN_FROM_URL>, newPassword: form.value.password })">
+    <!-- form fields -->
+    <button type="submit" :disabled="loading">Save Password</button>
+  </form>
+  <div>{{ result }} - Boolean confirmation of successful password change</div>
+</template>
+
+<script>
+import { useForgotPassword } from '@vsf-enterprise/commercetoolss';
+
+export default {
+  setup() {
+    const { reset, change, result, loading } = useForgotPassword();
+
+    return {
+      reset,
+      change,
+      result,
+      loading,
+    };
+  }
+};
+
+</script>
```

Comment:
I have some concerns about this code example:

1. I feel like we are mixing two things in one code block. Does it make sense to split this example into two parts - one example for generating token and the other for changing user password?
2. I'm not sure what `<TOKEN_FROM_URL>` is and where it comes from. Does calling `reset` triggers an email that contains a link with the token?
3. You are referring to the `form` object that isn't registered and resulted from the `setup` method.

Reply:
1. We have an example in authentication section in general docs about forgot password and there are two separate code blocks. That is why I wanted to do something different here that just adding path to `@vsf-enterprise/commercetools`.
2. Reset method does not trigger an email because we do not have an email provider integrated yet. We just call a GraphQL mutation that returns a token and we log it to the console for now to simulate real life example. I can change the TOKEN_FROM_URL to how it is implemented.
3. Yes, I wanted to not include too many things at once. But now when I look at it, it may look strange (Where does it come from?)